src/librustdoc/html/render.rs

```diff
@@ -2531,10 +2541,17 @@ fn render_deref_methods(w: &mut fmt::Formatter, cx: &Context, impl_: &Impl,
 fn render_impl(w: &mut fmt::Formatter, cx: &Context, i: &Impl, link: AssocItemLink,
                render_header: bool, outer_version: Option<&str>) -> fmt::Result {
     if render_header {
-        write!(w, "<h3 class='impl'><code>{}</code>", i.impl_)?;
-        let since = i.stability.as_ref().map(|s| &s.since[..]);
+        write!(w, "<h3 class='impl'><span class='in-band'><code>{}</code>", i.inner_impl())?;
+        let since = i.impl_item.stability.as_ref().map(|s| &s.since[..]);
         render_stability_since_raw(w, since, outer_version)?;
-        write!(w, "</h3>")?;
+        write!(w, "</span><span class='out-of-band'>")?;
+        if let Some(l) = href_for_item(&i.impl_item, cx) {
```

Comment:
Would it be possible to synthetically create an `Item` here or pass it through to keep this `href()` as an inherent method?